Remove Gazelle backend

zhztheplayer opened the pull request:
Gazelle backend has stopped maintenance. Removing all codes of it.

zhztheplayer commented:
@marin-ma

marin-ma commented:
There are some outdated specs in README.md
e.g. ~~We also implemented a basic backend using Arrow Computer Engine which is for reference only.~~
https://github.com/oap-project/gluten/blob/8fcc41c3987cff1dbacbce6a8a6003738e94c3d2/README.md?plain=1#L48

zhztheplayer commented:
> There are some outdated specs in README.md e.g. ~We also implemented a basic backend using Arrow Computer Engine which is for reference only.~
> 
> https://github.com/oap-project/gluten/blob/8fcc41c3987cff1dbacbce6a8a6003738e94c3d2/README.md?plain=1#L48

Fixed.
